src/Components/ActionsModal/ActionsModal.tsx

```diff
@@ -195,17 +192,21 @@ const ActionModal: React.FC<ActionModalProps> = ({
     }
     setData(defaultDecodedAction.decodedCall);
     setSelectedAction(action);
-    setSelectedActionContract(defaultDecodedAction.contract);
   }
 
   function saveSupportedAction(call?: DecodedCall) {
     const decodedCall = call ?? data;
-    if (!selectedAction || !decodedCall || !setSelectedActionContract) return;
+
+    const defaultDecodedAction = defaultValues[
+      decodedCall.callType
+    ] as DecodedAction;

```

Comment:
I would generally avoid using casting variables since it is bad TypeScript practice. Would it be feasible to either make a new interface or pick the interface from existing `DecodedAction`?

Reply:
Solved!

The issue was that the default data didn't contain all the needed parameters.